lib/spack/spack/version.py

```diff
@@ -882,33 +916,32 @@ def union(self, other):
 
     @coerced
     def intersection(self, other):
-        if self.overlaps(other):
-            if self.start is None:
-                start = other.start
-            else:
-                start = self.start
-                if other.start is not None:
-                    if other.start > start or other.start in start:
-                        start = other.start
+        if not self.overlaps(other):
+            return VersionList()
 
-            if self.end is None:
-                end = other.end
-            else:
-                end = self.end
-                # TODO: does this make sense?
-                # This is tricky:
-                #     1.6.5 in 1.6 = True  (1.6.5 is more specific)
-                #     1.6 < 1.6.5  = True  (lexicographic)
-                # Should 1.6 NOT be less than 1.6.5?  Hmm.
-                # Here we test (not end in other.end) first to avoid paradox.
-                if other.end is not None and end not in other.end:
-                    if other.end < end or other.end in end:
-                        end = other.end
-
-            return VersionRange(start, end)
+        if self.start is None:
+            start = other.start
+        else:
+            start = self.start
+            if other.start is not None:
+                if other.start > start or other.start in start:
+                    start = other.start
 
+        if self.end is None:
+            end = other.end
         else:
-            return VersionList()
+            end = self.end
+            # TODO: does this make sense?
```

Comment:
This simplifies if `self.next = next_version(self.end)` is stored, then it's just intersection of [x, y). For the example below with ranges :1.6.5 and :1.6 you can just store 1.6.6 and 1.7 and use <

Reply:
Anyways, for later... I remember I implemented that once.